components/formats-bsd/src/loci/formats/in/TileJPEGReader.java

```diff
@@ -122,34 +105,24 @@ public void initFile(String id) throws FormatException, IOException {
     super.initFile(id);
 
     in = new RandomAccessInputStream(id);
-    decoder = new JPEGTileDecoder();
-    decoder.initialize(in, 0, 1, 0);
+    JPEGTileDecoder decoder = new JPEGTileDecoder();
+    int[] dimensions = decoder.preprocess(in);
+    decoder.close();
```

Comment:
Makes sense, I just don't want to change ```JPEGTileDecoder``` here since it is removed in https://github.com/openmicroscopy/bioformats/pull/2883.  Happy to open a follow-up card to be addressed once ome-codecs settles down.